### PR TITLE
Move BuildCheck docs

### PR DESCRIPTION
**Pull request by JanKrivanek**
### Context
We are planning to ship the feature - so let's move specs from proposed space.

### Follow-up work need once merged
* [ ] Update https://aka.ms/buildcheck/codes (btw. is it the one you had in mind @baronfel? Or is there another as well?)
* [ ] Update docs link in https://github.com/dotnet/msbuild/issues/10372 (@f-alizada)

### CODE REVIEW

**Comment by baronfel** [comment]
The other location is the p6 release notes, but I can handle that

**Review by baronfel — APPROVED** [review]

File: documentation/specs/BuildCheck/Codes.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# `BuildCheck` reports codes and their meaning
+
+Report codes are chosen to conform to suggested guidelines. Those guidelines are currently in revew: https://github.com/dotnet/msbuild/pull/10088
+
+| Exit&nbsp;Code | Reason |
+|:-----|----------|
+| 0 | Success |
+| [BC0101](#BC0101) | Shared output path. |
+| [BC0102](#BC0102) | Double writes. |
```
Comment (f-alizada) [inline documentation/specs/BuildCheck/Codes.md:9]: This page seems to not have every code: https://github.com/dotnet/msbuild/blob/main/documentation/specs/proposed/BuildCheck/Codes.md

**Review by f-alizada — CHANGES_REQUESTED** [review]
Thank you for driving this! 
I would expect in this PR only this kind of changes: 
image
Please let me know if that is not the case, and if there are changes could you please point in the PR description what exacly was  updated?

**Comment by JanKrivanek** [comment]
@f-alizada thank you for eye for detail! I apparently had some stale branch. Move is fixed now

**Review by f-alizada — APPROVED** [review]
Thank you!

**Pull request merged**